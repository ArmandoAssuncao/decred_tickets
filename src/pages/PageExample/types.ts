import { IScreenProps } from 'containers/Root/types'
import { getUser } from 'store/user/actions'
import { IUserState } from 'store/user/types'

export interface IConnectedProps {
    user?: IUserState
}

export interface IActionCreators {
    getUser: typeof getUser
}

export interface IProps extends IScreenProps, IConnectedProps, IActionCreators {
}

export interface IState {
}
