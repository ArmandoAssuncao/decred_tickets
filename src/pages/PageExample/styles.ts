import { ViewStyle } from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'

export default EStyleSheet.create({
    container: {
        alignItems: 'center',
        backgroundColor: '$bgColor',
        flex: 1,
        justifyContent: 'center',
    } as ViewStyle,
})
