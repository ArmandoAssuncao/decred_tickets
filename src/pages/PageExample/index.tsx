import React from 'react'
import { View } from 'react-native'
import { NavigationScreenConfig } from 'react-navigation'
import { connect, MapDispatchToProps } from 'react-redux'
import { bindActionCreators } from 'redux'
import { IAppState } from 'store/types'

import i18n from 'config/i18n'
import { getUser } from 'store/user/actions'

import { IActionCreators, IConnectedProps, IProps, IState, } from './types'
import styles from './styles'

import { Text } from 'components'

const mapStateToProps = (state: IAppState): IConnectedProps =>
  ({
    user: state.user,
  })

const mapDispatchToProps = (dispatch: any): MapDispatchToProps<IActionCreators, IProps> =>
    bindActionCreators(
        {
            getUser,
        },
        dispatch
    )

export class PageExample extends React.Component<IProps, IState> {
    static navigationOptions: NavigationScreenConfig<any> = {
        title: i18n.t('pages.pageExample.title'),
    }

    constructor(props) {
        super(props)
    }

    componentDidMount() {
        this.props.getUser()
    }

    render(): JSX.Element {
        return(
            <View style={styles.container}>
                <Text id={'app.name'} m="default message" />
                <Text m="default message">Description</Text>
            </View>
        )
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(PageExample)
