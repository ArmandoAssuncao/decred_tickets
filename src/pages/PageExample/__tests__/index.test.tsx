import * as React from 'react'
import { shallowStore } from 'test/enzyme'

import PageExample from '../index'

describe('PageExample test render', () => {
    const store = {
        user: {
            error: false,
            name: 'name example',
        },
    }
    const wrapper = shallowStore(<PageExample getUser={jest.fn()} />, store)

    it('Testing snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('Testing instance of component', () => {
        expect(wrapper.instance()).toBeInstanceOf(PageExample.WrappedComponent)
    })

    it('has 2 Text', () => {
        expect(wrapper.find('Text')).toHaveLength(2)
    })
})
