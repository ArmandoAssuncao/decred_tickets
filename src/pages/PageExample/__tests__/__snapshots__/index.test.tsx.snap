// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageExample test render Testing snapshot 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#512DA8",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    id="app.name"
    m="default message"
  />
  <Text
    m="default message"
  >
    Description
  </Text>
</Component>
`;
